File: src/main/java/ru/examle/edu/controller/AuthController.java
package ru.examle.edu.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.*;
import ru.examle.edu.dto.JwtResponse;
import ru.examle.edu.dto.LoginRequest;
import ru.examle.edu.dto.RegisterRequest;
import ru.examle.edu.entity.User;
import ru.examle.edu.entity.enums.UserRole;
import ru.examle.edu.repository.UserRepository;
import ru.examle.edu.security.JwtUtils;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    @Autowired
    private AuthenticationManager authenticationManager;

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PasswordEncoder passwordEncoder;

    @Autowired
    private JwtUtils jwtUtils;

    @PostMapping("/login")
    public ResponseEntity<?> authenticateUser(@RequestBody LoginRequest loginRequest) {

        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(loginRequest.getEmail(), loginRequest.getPassword()));

        SecurityContextHolder.getContext().setAuthentication(authentication);
        String jwt = jwtUtils.generateToken(authentication);

        User userDetails = userRepository.findByEmail(loginRequest.getEmail()).orElseThrow();

        return ResponseEntity.ok(new JwtResponse(jwt,
                userDetails.getId(),
                userDetails.getEmail(),
                userDetails.getFullName()));
    }

    @PostMapping("/register")
    public ResponseEntity<?> registerUser(@RequestBody RegisterRequest signUpRequest) {
        if (userRepository.existsByEmail(signUpRequest.getEmail())) {
            return ResponseEntity
                    .badRequest()
                    .body("Error: Email is already in use!");
        }

        // Создаем нового пользователя
        User user = new User();
        user.setEmail(signUpRequest.getEmail());
        user.setPasswordHash(passwordEncoder.encode(signUpRequest.getPassword()));
        user.setFullName(signUpRequest.getFullName());
        user.setPosition(signUpRequest.getPosition());
        user.setDepartment(signUpRequest.getDepartment());
        
        // Устанавливаем роль по умолчанию
        user.setRole(UserRole.USER);

        userRepository.save(user);

        return ResponseEntity.ok("User registered successfully!");
    }
}

File: server/src/main/kotlin/com/nikshet/industrialbackend/contoller/AuthController.kt
package com.nikshet.industrialbackend.contoller

import com.nikshet.industrialbackend.domain.TokensEntity
import com.nikshet.industrialbackend.dto.request.LoginRequest
import com.nikshet.industrialbackend.exception.InvalidLoginOrPasswordException
import com.nikshet.industrialbackend.providers.JwtTokensProvider
import com.nikshet.industrialbackend.service.UsersService
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(
    private val usersService: UsersService,
    private val tokensProvider: JwtTokensProvider
) {

    @PostMapping("/login")
    fun login(@Valid @RequestBody request: LoginRequest): ResponseEntity<TokensEntity> {
        val user = usersService.authenticate(request.phoneNumber, request.password)
            ?: throw InvalidLoginOrPasswordException()

        val payload: Map<Any, Any> = mapOf(
            "sub" to user.id.toString(),
            "phone" to user.phoneNumber
        )

        val tokens = tokensProvider.generateTokensPairs(payload)
        return ResponseEntity.ok(tokens)
    }
}